pyradiosky/skymodel.py

```diff
@@ -491,7 +525,25 @@ def __init__(
                 self.Nfreqs = 1
 
             if reference_frequency is not None:
-                if not isinstance(reference_frequency, (Quantity,)):
+                if isinstance(reference_frequency, (list)):
+                    # try just converting the list to a Quantity. This will work if all
+                    # the elements are Quantities with compatible units or if all the
+                    # elements are just numeric (in which case the units will be "").
+                    warnings.warn(
+                        "reference_frequency is a list. Attempting to convert to a Quantity.",
+                    )
+                    try:
+                        reference_frequency = Quantity(reference_frequency)
+                    except (TypeError):
+                        raise ValueError(
+                            "If reference_frequency is supplied as a list, all the elements must be "
+                            "Quantity objects with compatible units."
+                        )
```

Comment:
There are at least two instances like this, is the TypeError thrown by astropy useful? should we be reraising it?

Reply:
If an item in the list is not a Quantity (e.g. just a float) the error is:
`TypeError: only dimensionless scalar quantities can be converted to Python scalars`
which isn't very helpful in my opinion. 
If items in the list are all Quantities but their units are not compatible then the error is:
`astropy.units.core.UnitConversionError: 'Jy' (spectral flux density) and 'Hz' (frequency) are not convertible`
That error message does seem useful, but it is not a TypeError, so it is not caught by the exception and you just get that astropy error. 

Now that I think about it, that error doesn't really tell you which parameter had the problem, so maybe I should add handling for that case by printing some text and then re-raising the original exception? Of course the stack trace would show the line in question, which would make which parameter is a problem clear.

Reply:
@mkolopanis Do you think leaving this alone, so that we get a more useful error when it's a TypeError and the UnitConversionError in that case is ok?

Reply:
oh yeah sorry. I read your last response and just never replied. I think your current set up works for me, it handles the case I was thinking in my head.